File: frontend/src/types/report.ts
// Report types
export interface ReportSummary {
  id: string;
  document_id: string;
  report_type: string;
  status: string;
  generated_at: string;
  download_url: string | null;
}

export interface ReportDetail extends ReportSummary {
  compliance_score: number;
  total_rules_checked: number;
  passed: number;
  failed: number;
  warnings: number;
  risk_score: number;
}

File: backend/app/schemas/report.py
"""Report schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ReportSummary(BaseModel):
    id: str
    document_id: str
    report_type: str
    status: str
    generated_at: datetime
    download_url: Optional[str] = None


class ReportDetail(ReportSummary):
    compliance_score: float
    total_rules_checked: int
    passed: int
    failed: int
    warnings: int
    risk_score: float
